packages/docusaurus/src/commands/build.ts

```diff
@@ -119,11 +119,14 @@ export default async function build(
 
   // Remove server.bundle.js because it is not needed.
   if (
-    serverConfig.output &&
-    serverConfig.output.filename &&
-    typeof serverConfig.output.filename === 'string'
+    serverConfig?.output &&
```

Comment:
I don't know why, but TypeScript was throwing a bunch of errors related to this piece of code, so I had to use optional chaining to get rid of them.

Reply:
this is surprising, do you remember the TS error?

Reply:
tried to remove these typing changes and the package build was successful this time. Strange, so reverted these changes.